docs/docs/guides/overview.md

```diff
@@ -1,8 +1,8 @@
-# Using DBSP
+# Using Feldera
```

Comment:
I think these can all stay "DBSP"

Reply:
This section should make it clear that these are not functionally equivalent.  One is a full-blown service, the other is a low-level engine.  And I agree that the latter shouldn't be called Feldera in this context.  The SQL tutorial also needs work, as it doesn't really explain how to use Feldera, it's more of an internal architecture document. But that's outside the scope of this PR.